pypi_formula_mappings.json

```diff
@@ -766,9 +766,6 @@
       "typing-extensions"
     ]
   },
-  "pgxnclient": {
-    "exclude_packages": ["six"]
-  },
```

Comment:
We need to add `setuptools` as an extra package here, otherwise it'll get removed on the next version bump.

Reply:
Why is that? I did not know that. I though this file was about mappings between deps and resources; not some workaround against resource removals?

Reply:
In this case `setuptools` isn't listed as a dep explicitly by upstream:
```
╰─ pipgrip --sort pgxnclient
pgxnclient==1.3.2
six==1.16.0
```
 Since we are on Python 3.12 we need to add it ourselves since that's no longer bundled. Kind of like how we handle it for the `python@3.*` formulae themselves:
https://github.com/Homebrew/homebrew-core/blob/f1e245ba9a75aea2567964e5ac56de7ecd311110/pypi_formula_mappings.json#L850-L864

Reply:
@p-linnane it only uses `setuptools` as a PEP 517 build backed. it's not a runtime dependency, so no resource required here

previously, this needed a `depends_on "python-setuptools" => :build` because `std_pip_args` [disables build isolation](https://github.com/Homebrew/brew/blob/5d3fae3571c034896c265e9c3cef5e1b84edb37f/Library/Homebrew/formula.rb#L1692) (meaning `pip` won't install any required backends) whereas our `virtualenv_install_with_resources` [does not](https://github.com/Homebrew/brew/blob/5d3fae3571c034896c265e9c3cef5e1b84edb37f/Library/Homebrew/language/python.rb#L325)

Reply:
Thank you for the context. I wasn't aware. So dropping the resource will get us where we need to be then.

Reply:
fixed